R/get_boxi_extract_path.R

```diff
@@ -13,38 +13,42 @@
 get_boxi_extract_path <-
   function(year,
            type = c(
-             "Acute",
-             "MH",
-             "Maternity",
-             "Outpatient",
              "AE",
              "AE_CUP",
+             "Acute",
+             "CMH",
+             "Deaths",
              "DN",
              "GP_OoH-c",
              "GP_OoH-d",
              "GP_OoH-o",
-             "Deaths",
-             "CMH",
-             "Homelessness"
+             "Homelessness",
+             "Maternity",
+             "MH",
+             "Outpatients"
            )) {
     type <- match.arg(type)
 
     year_dir <- get_year_dir(year, extracts_dir = TRUE)
 
+    if (!check_year_valid(year, type)) {
+      return(NA)
+    }
+
```

Comment:
I think we should also just put this function / code block in `get_source_extract_path()` now.